apps/web/src/components/Dropdown/index.tsx

```diff
@@ -6,57 +7,91 @@ import {
   Transition,
 } from "@headlessui/react";
 import { ChevronUpDownIcon, XMarkIcon } from "@heroicons/react/24/solid";
+import cn from "classnames";
 
+import useOverflow from "~/hooks/useOverflow";
 import { Option } from "./Option";
 
 export interface Option {
   value: string | number;
   label?: ReactNode;
   prefix?: ReactNode;
+  inputDisplay?: ReactNode;
 }
+
 export interface DropdownProps {
   options: Option[];
-  selected?: Option | null;
   width?: string;
   placeholder?: string;
   clearable?: boolean;
-  onChange(newOption: Option | null): void;
+  selected?: Option | Option[] | null;
+  multiple?: boolean;
+  onChange(newOption: Option | Option[] | null): void;
 }
 
 const DEFAULT_WIDTH = "w-32";
 
 export const Dropdown: React.FC<DropdownProps> = function ({
   options,
   selected,
+  multiple,
   width,
   onChange,
   clearable = false,
   placeholder = "Select an item",
 }) {
+  const hasValue = Array.isArray(selected) ? selected.length > 0 : selected;
+
+  const containerRef = useRef<HTMLDivElement | null>(null);
+  const innerRef = useRef<HTMLDivElement | null>(null);
+  const isOverflowing = useOverflow(containerRef, innerRef);
+
   return (
-    <Listbox value={selected} onChange={onChange}>
+    <Listbox value={selected} onChange={onChange} multiple={multiple}>
       <div className="relative">
         <ListboxButton
           className={`relative h-9 ${
             width ?? DEFAULT_WIDTH
           } flex cursor-pointer items-center justify-between rounded-lg border border-transparent bg-controlBackground-light pl-2 pr-8 text-left text-sm shadow-md hover:border hover:border-controlBorderHighlight-light focus:outline-none focus-visible:border-indigo-500 focus-visible:ring-2 focus-visible:ring-white active:border-controlBorderHighlight-dark ui-open:border-controlActive-light dark:bg-controlBackground-dark dark:hover:border-controlBorderHighlight-dark dark:ui-open:border-controlActive-dark`}
         >
-          <div className="truncate align-middle">
-            {selected ? (
-              selected.label ?? selected.value
+          <div
```

Comment:
Can we add horizontal scrolling to allow users to navigate through all selected options? Currently, when multiple options are selected, only the ones that fit within the visible area are displayed, while the overflowed options are hidden.

Reply:
Done 👍